src/binary/non_blocking/binary_buffer.rs

```diff
@@ -262,7 +262,7 @@ impl<A: AsRef<[u8]>> BinaryBuffer<A> {
     ///
     /// See: https://amzn.github.io/ion-docs/docs/binary.html#uint-and-int-fields
     pub fn read_uint(&mut self, length: usize) -> IonResult<DecodedUInt> {
-        if length <= mem::size_of::<usize>() {
+        if length <= mem::size_of::<u64>() {
```

Comment:
🗺️ Here and below: the code was originally checking for `usize`, which had the potential to be overly conservative. On 32-bit systems, this would cause medium-sized values to be pushed into a `BigInt`.

In some cases (like reading a `UInt` that represents a `SymbolId`), limiting it to `usize` makes sense. However, that's not this method's responsibility.